### PR TITLE
raft: make empty learner accept snapshots

### PR DESCRIPTION
**Pull request by gyuho**
If learner just has been added by conf change, it should accept leader snapshot.

Fix https://github.com/etcd-io/etcd/issues/9498.
Replace https://github.com/etcd-io/etcd/pull/9709.

### CODE REVIEW

File: raft/raft_test.go [hunk 1]
```diff
@@ -2914,6 +2914,44 @@ func TestLearnerReceiveSnapshot(t *testing.T) {
 	}
 }
 
+// TestNormalPeerRejectLearnerSnapshot tests that normal peer should
```
Comment (xiang90) [inline raft/raft_test.go]: shouldnt we test `If learner just has been added by conf change, it should accept leader snapshot.` instead?

**Comment by gyuho** [comment]
Believe this is not needed since https://github.com/etcd-io/etcd/commit/b171e1c78b0411c8a2648de0eee2102735bec6b4#diff-a8629a0eb6da1c6c3805111f1f1af85eR1331.

**Pull request closed without merging**